### PR TITLE
Fix role-based authorization in Clients section.

### PR DESCRIPTION
**Pull request by ssilvert**
## Motivation
The PR fixes role-based authorization in the Clients section by hiding components or making them read-only as needed.

## Brief Description
This is in preparation for implementing fine-grained authorization, where flags such as `isManager` will also include a check for fine-grained authorization.

## Verification Steps
To verify the PR, it is best to log into a test realm with a test user.  Then set limited client roles for that user on the realm-management client.

To see the Clients section the user should have "view-clients" and "query-clients" roles.
To see the Permissions tab, the user also needs "manage-authorization" role.
To see Keys and Credentials tabs, the client needs to have `Client authentication` enabled.
To see Authorization tab, the client needs to have `Authorization` enabled.
To see Service account roles tab, the client needs to have `Service account roles` enabled.

### CODE REVIEW

File: src/clients/authorization/Settings.tsx [hunk 1]
```diff
@@ -166,6 +166,7 @@ export const AuthorizationSettings = ({ clientId }: { clientId: string }) => {
           name="authenticationSettings"
           save={() => handleSubmit(save)()}
           reset={() => reset(resource)}
+          isActive={true}
```
Comment (jonkoops) [inline src/clients/authorization/Settings.tsx:169]: For boolean props that are true the value can be omitted.
```suggestion
          isActive
```

File: src/components/role-mapping/RoleMapping.tsx [hunk 2]
```diff
@@ -253,36 +279,23 @@ export const RoleMapping = ({
...
-              return false;
-            },
-          },
-        ]}
+        actions={
+          isManager
+            ? [
+                {
+                  title: t("common:unAssignRole"),
+                  onRowClick: async (role) => {
```
Comment (jonkoops) [inline src/components/role-mapping/RoleMapping.tsx:290]: There is nothing asynchronous happening here, so I believe you can omit the `async` keyword.
```suggestion
                  onRowClick: (role) => {
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: src/components/role-mapping/RoleMapping.tsx (hunk 2 above)
Comment (ssilvert) [reply src/components/role-mapping/RoleMapping.tsx:290]: That was not my code, but I think whoever wrote it was correct.  When I remove the `async` I get `Type 'boolean' is not assignable to type 'void | Promise<boolean>`

File: src/clients/authorization/Settings.tsx (hunk 1 above)
Comment (ssilvert) [reply src/clients/authorization/Settings.tsx]: fixed

File: src/clients/ClientsSection.tsx [hunk 3]
```diff
@@ -117,29 +149,7 @@ export default function ClientsSection() {
...
-                        <Link {...props} to={toImportClient({ realm })} />
-                      )}
-                      variant="link"
-                    >
-                      {t("importClient")}
-                    </Button>
-                  </ToolbarItem>
-                </>
-              }
+              toolbarItem={ToolbarItems()}
```
[the remark this answers is not in the document]
Comment (ssilvert) [reply src/clients/ClientsSection.tsx]: fixed

File: src/components/role-mapping/RoleMapping.tsx (hunk 2 above)
Comment (jonkoops) [reply src/components/role-mapping/RoleMapping.tsx:290]: Ah, yes. That's fine then.

**Review by jonkoops — APPROVED** [review]

**Pull request merged**